src/qz/utils/ConnectionUtilities.java

```diff
@@ -0,0 +1,29 @@
+package qz.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+
+import qz.common.Constants;
+
+public final class ConnectionUtilities {
+    private ConnectionUtilities() {
+    }
+
+    public static InputStream getInputStream(String urlString) throws IOException {
+        URLConnection urlConn = new URL(urlString).openConnection();
+        urlConn.setRequestProperty("User-Agent", userAgent());
+        return urlConn.getInputStream();
+    }
+
+    public static String userAgent() {
+        String.format("%s/%s (%s; %s) Java/%s (%s)",
+            Constants.ABOUT_TITLE.replaceAll("[^a-zA-Z]", ""),
+            Constants.VERSION,
```

Comment:
Since most user-agent strings use `major.minor`, I would recommend truncated off the `.patch` and just having it `QZTray/2.1`.